### DISCUSSION TITLE
getTrajectory sums traj value across breaks instead of mean?

### DISCUSSION THREAD

**Discussion by Zepeng-Mu** [GreenleafLab/ArchR#1291] [Questions / Documentation]
Hello, I'm trying to plot "MotifMatrix" value along a trajectory, which is similar to pueudotime but not exactly the same.
I found that in `getTrajectory`, trajectory values are divided into 100 bins, and this comes as natural because trajectory value is between 1 and 100. Then the matrix values in these bins are sum together, and this is done by `.getGroupMatrix` function. I think this is a bit problematic in my case, because in my data 80% of the cells have a trajectory value below 10, and the rest are distributed between 10 and 100 for instance. In this case, even if a matrix value is correlated with trajectory score, it may not be the case after summing and smoothing, because the number of cells in each bin (therefore summation) is biased towards those with low trajectory scores.

I think at least in my case, getting the mean instead of sum would make more sense. I'm wondering whether it's possible to have this feature in the package.

Thanks so much!

**Comment by rcorces** [comment]
Unless I'm mis-understanding your post, this would normally be handled by the `scaleTo` argument? Though in the case of motif deviations, ArchR expects you to not use either scaleTo or log2Norm

**Reply by Zepeng-Mu** [reply]
Hi,
Sorry I was not clear enough. May be it's better for me to include some examples.
Currently I have a "score" that's correlated with naive T cell, for example, and this is similar to pseudotime but not the same as pseudotime. I want to visualize it's correlation with TF deviation Z score.

![image](https://user-images.githubusercontent.com/21952317/154338257-12c7d598-c1d2-4f9d-8362-70555d8190ea.png)

The results looks like this. I used `scaleTo` but the figure looks the same when setting it to `NULL`. Both LEF1 and TCF7 show positive correlation with my score here, which makes sense. But on the left side of the heatmap there is this strip with very dark blue and red colors. The reason I think is because there are too many cells in those bins whose scores are summed up.
![Screen Shot 2022-02-16 at 13 14 38](https://user-images.githubusercontent.com/21952317/154338814-29b9d272-f9bf-4771-9cda-40f4efc73dba.png)

Here, for example, I plotted deviation of LEF1 against my naive T cell score. By density plot we can see a huge number of cells with deviation around 0. The mean deviations of these cells would be around 0, but their sum could be very negative. And this results to the really dark blue band on the left side of the heatmap above.
